### ISSUE TITLE
Enable viewing inline images in channel

### ISSUE THREAD

**Issue by naclander** [jorgenschaefer/circe#316]
Some IRC clients show the image inline when a link is pasted. For example:
https://i.stack.imgur.com/CWP6e.png

Emacs supports showing images, and so it would be great if Circe support this. 

Also, thanks for working on this great IRC client so far.

**Comment by jorgenschaefer** [comment]
Nice idea. It should be optional, and there should be a way of hiding an image once displayed (some of them are obnoxious etc.). Might also be necessary to scale the image at times?

**Comment by naclander** [comment]
I agree it should definitely be optional. A toggle-image function would be nice to have as well.

We should definitely also have a cap-image-scaling variable to limit the image to some maximum value ( images smaller than this maximum value should display without an issue ).

Thanks for responding, I might attempt to implement a prototype if time permits.

**Comment by wasamasa** [comment]
I considered a similar thing, but when hovering over a link.

**Comment by naclander** [comment]
@jorgenschaefer How did you envision hiding an image once its displayed? I have a working implementation of eveything except that particular feature.

Were you thinking toggling off all images globally? Was it on an image-by-image basis? If so, how will the user select a specifci image to disable? I could see them calling a functino with the image-url as the argument, and then removing that particular image.

**Comment by wasamasa** [comment]
How about having a key binding that hides the image at point (or better, supports toggling that)?

**Comment by naclander** [comment]
I like that solution, I'll try and implement it.

**Comment by naclander** [comment]
Actually, url-at-point might end up being better. If we end up using image-at-point ( the actual string with the display property ), once you toggle it off, it will be hard ( impossible ) to find the image again and toggle it back on, because the image is now invisible.

**Comment by naclander** [comment]
#320 implements this. Let me know if it works for you guys.

**Comment by alphapapa** [comment]
@naclander Nicely done!  I just implemented similar functionality for matrix-client: https://github.com/jgkamat/matrix-client-legacy-el/pull/10  I noticed that you passed the type `'imagemagick` to `create-image`.  I can't find that in docs anywhere.  Can you point me to where you found that?  I think that might help me too.

**Comment by naclander** [comment]
Thanks @alphapapa, you can read about imagemagick support [here](https://www.gnu.org/software/emacs/manual/html_node/elisp/ImageMagick-Images.html)

**Comment by alphapapa** [comment]
Thank you very much!

**Comment by jorgenschaefer** [comment]
I guess this is fixed now? Thank you for the nice work!

**Comment by naclander** [comment]
@jorgenschaefer this is not fixed, and is pending #320 

I'm actually waiting on your to comment on some of my responses.

**Comment by jorgenschaefer** [comment]
Sorry, responded there.
